src/classes/tmDaoProvisioning.cls

```diff
@@ -0,0 +1,174 @@
+        sfbase__CountryRevenueMap__c revMapping = getcountryRevMap().get(countryIsoCode);
+        if (revMapping != null) {
+            String revOwner = revMapping.sfbase__RevenueOwner__c;        
+
+            List<sfbase__SOBPaymentMethodMap__c> paymentMethodMapping = [SELECT sfbase__DefaultPaymentTerm__c
+                                                        FROM sfbase__SOBPaymentMethodMap__c
+                                                        WHERE sfbase__PaymentMethod__c = :tmConstants.CONTRACT_PAYMENT_METHOD
+                                                        AND sfbase__RevenueOwner__c = :revOwner
+                                                        LIMIT 1];
+            if (paymentMethodMapping.size() > 0) {
+                paymentTerm = paymentMethodMapping.get(0).sfbase__DefaultPaymentTerm__c;
+            }                                                   
+        }
+        return paymentTerm;
+    }  
+
+    public static Id createOrderOnNewContract(Contract c, Order ord, List<OrderItem> ordItems, Id priceBookId) {
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(c, ord, ordItems, priceBookId);
+    }
+
+    public static Id createOrderOnExistingContract(Id contractId, Order ord, List<OrderItem> ordItems) {
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(contractId, ord, ordItems);
+    }
+    public static id upgradeOrderOnExistingContract(Id contractId, Order ord, List<OrderItem> ordItems, List<OrderItem> oldOrdItems) {
+        provisioningHelper.updateOrderItems(oldOrdItems);
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(contractId, ord, ordItems);
+    }
+    private static tmDaoProvisioningHelper provisioningHelper {
+        get {
+            if (provisioningHelper == null) {
+                provisioningHelper = new tmDaoProvisioningHelper();
+            }
+            return provisioningHelper;
+        }
+        private set;
+    }
+
+    private without sharing class tmDaoProvisioningHelper {
+        public Id doCreateOrderOnNewContractWithoutSharing(Contract c, Order ord, List<OrderItem> ordItems, Id pbId) {
+            c.RecordTypeId = recordTypeIds.get('Contract');
+            insert c;
+
+            //Activate contract
+            c.Status = tmConstants.CONTRACT_STATUS_ACTIVATED;
+            update c;
+
+            Id orderId = doCreateOrderOnNewContractWithoutSharing(c.Id, ord, ordItems);
+            return orderId;
+        }
+
+        public Id doCreateOrderOnNewContractWithoutSharing(Id contractId, Order ord, List<OrderItem> ordItems) {
+            ord.contractId = contractId;
+            ord.RecordTypeId = recordTypeIds.get('Order');
+
+            //1.Insert order
+            insert ord;
+            Id orderId = ord.Id;
+
+            //2.Insert order items
+            for (OrderItem oi : ordItems) {
+                oi.OrderId = orderId;
+            }
+            insert ordItems;
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.